services/workflows-service/src/business-report/business-report.controller.external.ts

```diff
@@ -224,9 +259,32 @@ export class BusinessReportControllerExternal {
     @CurrentProject() currentProjectId: TProjectId,
     @Param('id') id: string,
   ) {
-    const { id: customerId } = await this.customerService.getByProjectId(currentProjectId);
+    const { id: customerId, features } = await this.customerService.getByProjectId(
+      currentProjectId,
+    );
+
+    const report = await this.businessReportService.findById({ id, customerId });
+    const business = await this.businessService.getById(
+      report.merchantId,
+      { select: { metadata: true } },
+      [currentProjectId],
+    );
+
+    const metadata = business?.metadata as {
+      featureConfig?: TCustomerWithFeatures['features'];
+    };
+
+    const isOngoingEnabledForBusiness =
+      metadata?.featureConfig?.[FEATURE_LIST.ONGOING_MERCHANT_REPORT]?.enabled;
 
-    return await this.businessReportService.findById({ id, customerId });
+    return {
+      ...report,
+      monitoringStatus:
+        (isOngoingEnabledForBusiness ||
+          (isOngoingEnabledForBusiness === undefined &&
+            features?.ONGOING_MERCHANT_REPORT?.options?.runByDefault)) ??
+        false,
```

Comment:
I think this would be easier to read if the conditions were separated or if if conditions were in use.